### PR TITLE
Check for nil optional prameter in List

### LINKED ISSUE #2018: PL PANIC on GET to /containers/list
I wanted to see the values being returned via `List()` of containers.  This apparently PANICs.  

Swagger spec says this should be possible
```
1004 »·······»·······"/containers/list": {
1005 »·······»·······»·······"get": {
1006 »·······»·······»·······»·······"description": "Gets a list of all containers",
1007 »·······»·······»·······»·······"operationId": "GetContainerList",
1008 »·······»·······»·······»·······"tags": [
1009 »·······»·······»·······»·······»·······"containers"
1010 »·······»·······»·······»·······],
1011 »·······»·······»·······»·······"consumes": [
1012 »·······»·······»·······»·······»·······"application/octet-stream"
1013 »·······»·······»·······»·······],
1014 »·······»·······»·······»·······"produces": [
1015 »·······»·······»·······»·······»·······"application/json"
1016 »·······»·······»·······»·······],
1017 »·······»·······»·······»·······"parameters": [
1018 »·······»·······»·······»·······»·······{
1019 »·······»·······»·······»·······»·······»·······"name": "all",
1020 »·······»·······»·······»·······»·······»·······"required": false,
1021 »·······»·······»·······»·······»·······»·······"in": "query",
1022 »·······»·······»·······»·······»·······»·······"type": "boolean"
1023 »·······»·······»·······»·······»·······}
1024 »·······»·······»·······»·······],
1025 »·······»·······»·······»·······"responses": {
1026 »·······»·······»·······»·······»·······"200": {
1027 »·······»·······»·······»·······»·······»·······"description": "OK",
1028 »·······»·······»·······»·······»·······»·······"schema": {
1029 »·······»·······»·······»·······»·······»·······»·······"type": "array",
1030 »·······»·······»·······»·······»·······»·······»·······"items": {
1031 »·······»·······»·······»·······»·······»·······»·······»·······"$ref": "#definitions/ContainerListInfo"
1032 »·······»·······»·······»·······»·······»·······»·······}
1033 »·······»·······»·······»·······»·······»·······}
1034 »·······»·······»·······»·······»·······},
1035 »·······»·······»·······»·······»·······"500": {
1036 »·······»·······»·······»·······»·······»·······"description": "server error",
1037 »·······»·······»·······»·······»·······»·······"schema": {
1038 »·······»·······»·······»·······»·······»·······»·······"$ref": "#/definitions/Error"
1039 »·······»·······»·······»·······»·······»·······}
1040 »·······»·······»·······»·······»·······}
1041 »·······»·······»·······»·······}
1042 »·······»·······»·······}
1043 »·······»·······},
```

```
$ curl -k https://192.168.8.131:2378/logs/port-layer.log
```

```
2016/08/17 00:03:09 http: panic serving 192.168.8.131:40018: runtime error: invalid memory address or nil pointer dereference
goroutine 462 [running]:
net/http.(*conn).serve.func1(0xc420174600)
        /usr/local/go/src/net/http/server.go:1491 +0x12a
panic(0xde8fa0, 0xc42000e050)
        /usr/local/go/src/runtime/panic.go:458 +0x243
github.com/vmware/vic/lib/portlayer/exec.convertInfraContainers(0xc42067d000, 0x2, 0x2, 0x0, 0x2, 0x2, 0x0)
        /go/src/github.com/vmware/vic/lib/portlayer/exec/container.go:597 +0x79
github.com/vmware/vic/lib/portlayer/exec.List(0x7fd5afbd5848, 0xc42000e468, 0xc42024d4f0, 0x0, 0xc42000e468, 0x71, 0xecf45a0bd, 0x2a01e6d5, 0x1956f80)
        /go/src/github.com/vmware/vic/lib/portlayer/exec/container.go:521 +0x7e
github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*ContainersHandlersImpl).GetContainerListHandler(0xc42042a088, 0xc4207b4e10, 0x0, 0x0, 0x0)
        /go/src/github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers/containers_handlers.go:259 +0x127
github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*ContainersHandlersImpl).GetContainerListHandler-fm(0xc4207b4e10, 0x0, 0x30003c420534901, 0xc4203cf3f0)
        /go/src/github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers/containers_handlers.go:55 +0x3e
github.com/vmware/vic/lib/apiservers/portlayer/restapi/operations/containers.GetContainerListHandlerFunc.Handle(0xc4206e1330, 0xc4207b4e10, 0x0, 0x1923460, 0xc4203cf3f0)
        /go/src/github.com/vmware/vic/lib/apiservers/portlayer/restapi/operations/containers/get_container_list.go:17 +0x3a
github.com/vmware/vic/lib/apiservers/portlayer/restapi/operations/containers.(*GetContainerList).ServeHTTP(0xc4202c8860, 0x1930260, 0xc4206eadd0, 0xc4207b4e10)
        /go/src/github.com/vmware/vic/lib/apiservers/portlayer/restapi/operations/containers/get_container_list.go:49 +0x18f
github.com/vmware/vic/vendor/github.com/go-swagger/go-swagger/httpkit/middleware.newOperationExecutor.func1(0x1930260, 0xc4206eadd0, 0xc4207b4e10)
        /go/src/github.com/vmware/vic/vendor/github.com/go-swagger/go-swagger/httpkit/middleware/operation.go:23 +0x69
net/http.HandlerFunc.ServeHTTP(0xc420418d30, 0x1930260, 0xc4206eadd0, 0xc4207b4e10)
        /usr/local/go/src/net/http/server.go:1726 +0x44
github.com/vmware/vic/vendor/github.com/go-swagger/go-swagger/httpkit/middleware.newRouter.func1(0x1930260, 0xc4206eadd0, 0xc4207b4e10)
        /go/src/github.com/vmware/vic/vendor/github.com/go-swagger/go-swagger/httpkit/middleware/router.go:78 +0x555
net/http.HandlerFunc.ServeHTTP(0xc4205dc100, 0x1930260, 0xc4206eadd0, 0xc4207b4e10)
        /usr/local/go/src/net/http/server.go:1726 +0x44
github.com/vmware/vic/vendor/github.com/go-swagger/go-swagger/httpkit/middleware.specMiddleware.func1(0x1930260, 0xc4206eadd0, 0xc4207b4e10)
        /go/src/github.com/vmware/vic/vendor/github.com/go-swagger/go-swagger/httpkit/middleware/spec.go:36 +0x7b
net/http.HandlerFunc.ServeHTTP(0xc4205de6e0, 0x1930260, 0xc4206eadd0, 0xc4207b4e10)
        /usr/local/go/src/net/http/server.go:1726 +0x44
net/http.serverHandler.ServeHTTP(0xc42080df00, 0x1930260, 0xc4206eadd0, 0xc4207b4e10)
        /usr/local/go/src/net/http/server.go:2202 +0x7d
net/http.(*conn).serve(0xc420174600, 0x1930de0, 0xc4206cbbc0)
        /usr/local/go/src/net/http/server.go:1579 +0x4b7
created by net/http.(*Server).Serve
        /usr/local/go/src/net/http/server.go:2293 +0x44d
```

### PR DESCRIPTION
**Pull request by fdawg4l**
Optional params may be nil.  Check for nil before blindly trying to deference a pointer.

Fixes #2018

### CODE REVIEW

File: lib/portlayer/exec/container.go [hunk 1]
```diff
@@ -517,8 +517,14 @@ func List(ctx context.Context, sess *session.Session, all *bool) ([]*Container,
 	if err != nil {
 		return nil, err
 	}
+
 	// convert to container
-	containers := convertInfraContainers(moVMs, all)
+	var t bool
+	if all != nil {
```
Comment (hmahmood) [inline lib/portlayer/exec/container.go]: Why is `all` even a pointer here?

Comment (fdawg4l) [inline lib/portlayer/exec/container.go]: Swagger represents optional vars as pointers so they can be set to `nil` if omitted.

**Comment by caglar10ur** [comment]
LGTM :sheep: :it:

**Comment by fdawg4l** [comment]
.

**Comment by fdawg4l** [comment]
poke @cgtexmex

**Comment by cgtexmex** [comment]
all is a boolean provided by the docker client -- if it was not provided I would expect it to panic here: 

https://github.com/vmware/vic/blob/master/lib/apiservers/engine/backends/container.go#L953

Were you able to recreate this?  I attempted a few times and had no success.  Ultimately this code will be going away or refactored for restart, so if you think this is it go for it.

**Comment by fdawg4l** [comment]
@cgtexmex `Config` isn't null there so it won't panic there. 

Yes, I can recreate it consistently using curl (see above for invocation).  And I verified my fix does fix it.

**Comment by cgtexmex** [comment]
LGTM

**Comment by fdawg4l** [comment]
.

**Pull request merged**